test/src/sender/main.cpp

```diff
@@ -16,51 +25,86 @@
 #include "../../../lib/vendor/ndtime.hpp"
 #include "../../../lib/vendor/iestream.hpp"
 #include "../../../include/data_structures/message.hpp"
-#include "../../../include/atomics/sendercadmium.hpp"
+#include "../../../include/atomics/sender_cadmium.hpp"
 
 using namespace std;
 using hclock = chrono::high_resolution_clock;
 using TIME = NDTime;
 
-/***** SETING INPUT PORTS FOR COUPLEDs *****/
+/**
+* Setting up in_ports for messages 
+*/
+
 struct inp_controll : public cadmium::in_port<message_t> {
 };
 struct inp_ack : public cadmium::in_port<message_t> {
 };
 
-/***** SETING OUTPUT PORTS FOR COUPLEDs *****/
+/**
+* Setting up out_ports for messages 
+*/
+
 struct outp_ack : public cadmium::out_port<message_t> {
 };
 struct outp_data : public cadmium::out_port<message_t> {
 };
 struct outp_pack : public cadmium::out_port<message_t> {
 };
 
-/********************************************/
-/****** APPLICATION GENERATOR *******************/
-/********************************************/
+/**
+ * The application generator is the class that takes respective file path
+ * parameter and waits for input
+ * @tParam message T
+ */
 template<typename T>
 class ApplicationGen : public iestream_input<message_t, T> {
     public:
         ApplicationGen() = default;
+        /**
+         * This is a contructor with parameters for the class of application 
+         * generator takes the input file path which contains 
+         * the input that is fed in to the application to execute
+         * @param file_path
+         */
         ApplicationGen(const char* file_path) : iestream_input<message_t, T>(
             file_path
         ) {}
 };
 
 int main() {
-    auto start = hclock::now(); //to measure simulation execution time
+    /**
+     * to measure simulation execution time
+     */
+    auto start = hclock::now(); 
 
-    /*************** Loggers *******************/
+    /**
+     * Generating messages and operation logs which are being fed
+     * at execution time of this application and storing them 
+     * in the respective path as mentioned to sender_test_output 
+     * file as indicated in the out_data.
+     */
     static std::ofstream out_data(
         "../test/data/sender/sender_test_output.txt"

```

Comment:
File path here is hardcoded which need to be softcoded using #define. Kindly make the changes in all the .cpp files (top_model, receiver, sender, subnet).

Reply:
Hi,

Updated, thanks. Please review and approve the request.

Thanks